testsuites/sparql/src/main/java/org/eclipse/rdf4j/testsuite/sparql/tests/FilterScopeTest.java

```diff
@@ -0,0 +1,267 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eclipse RDF4J contributors.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ ******************************************************************************/
+
+package org.eclipse.rdf4j.testsuite.sparql.tests;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.rdf4j.model.BNode;
+import org.eclipse.rdf4j.model.util.Values;
+import org.eclipse.rdf4j.model.vocabulary.FOAF;
+import org.eclipse.rdf4j.model.vocabulary.RDF;
+import org.eclipse.rdf4j.query.BindingSet;
+import org.eclipse.rdf4j.query.TupleQueryResult;
+import org.eclipse.rdf4j.query.explanation.Explanation;
+import org.eclipse.rdf4j.testsuite.sparql.AbstractComplianceTest;
+import org.junit.Test;
+
+/**
+ * Tests for variable scoping issues with SPARQL FILTER clauses.
+ *
+ * @author Håvard M. Ottestad
+ */
+public class FilterScopeTest extends AbstractComplianceTest {
+
+	@Test
+	public void testScope1() {
+		loadData();
+
+		String query = String.join("\n", "",
+				"PREFIX foaf: <" + FOAF.NAMESPACE + ">",
+				"SELECT * WHERE {",
+				"  ?person a ?type ; ",
+				"  foaf:age ?age ;",
+				"  foaf:knows ?person2 .",
+				"  FILTER(?person != <http://example.org/2> && ?type = foaf:Person)",
+				"	{",
+				"      ?person foaf:age ?age .",
+				"      FILTER(?type NOT IN (foaf:NotPerson)) #?type is not in scope, so should produce an error. ",
```

Comment:
Yep - the error here that should happen is a type error when evaluating the `NOT IN` operator, since its operand is unbound. The net result of this should be that the `FILTER` condition is effectively always `false`.